conan/tools/microsoft/msbuild.py

```diff
@@ -48,11 +68,83 @@ def command(self, sln, targets=None):
+        cmd = self.command(sln, targets=targets, force_import_generated_files=force_import_generated_files)
         self._conanfile.run(cmd)
 
     @staticmethod
     def get_version(_):
         return NotImplementedError("get_version() method is not supported in MSBuild "
                                    "toolchain helper")
+
+    def _get_concrete_props_file(self, root_props_file):
+        concrete_props_file = ""
+
+        root = ET.parse(root_props_file).getroot()
+        namespace = re.match('\{.*\}', root.tag)
+        namespace = namespace.group(0) if namespace else ""
+        importgroup_element = root.find(f"{namespace}ImportGroup")
+        if importgroup_element:
+            import_elements = importgroup_element.findall(f"{namespace}Import")
+            if len(import_elements) == 1:
+                concrete_props_file = import_elements[0].attrib.get("Project")
+            else:
+                expected_condition = (f"'$(Configuration)' == '{self.configuration}' And "
+                                      f"'$(Platform)' == '{msbuild_arch_to_conf_arch(self.platform)}'")
+                for import_element in import_elements:
+                    if expected_condition == import_element.attrib.get("Condition"):
+                        concrete_props_file = import_element.attrib.get("Project")
+                        break
+
+        if concrete_props_file:
+            concrete_props_file = os.path.join(self._conanfile.generators_folder, concrete_props_file)
+
+        if not concrete_props_file or not os.path.exists(concrete_props_file):
+            raise ConanException(
+                f"MSBuildToolchain props file is missing for configuration={self.configuration} and "
+                f"platform={msbuild_arch_to_conf_arch(self.platform)}."
+            )
+
+        return concrete_props_file
+
+    def _get_msbuildtoolchain_properties(self, root_props_file):
+        properties = {}
+
+        # Get properties from props file of configuration and platform
+        concrete_props_file = self._get_concrete_props_file(root_props_file)
+        root = ET.parse(concrete_props_file).getroot()
+        namespace = re.match('\{.*\}', root.tag)
+        namespace = namespace.group(0) if namespace else ""
+        for propertygroup in root.iter(f"{namespace}PropertyGroup"):
+            if propertygroup.attrib.get("Label") == "Configuration":
+                for child in propertygroup:
+                    propert_name = child.tag.replace(namespace, "")
+                    properties[propert_name] = child.text
+        return properties
+
+    def _force_import_generated_files_cmd_line_arg(self):
+        cmd_args = []
+        props_paths = []
+
+        # MSBuildToolchan must be in generators for this MSBuild mode
+        msbuildtoolchain_file = os.path.join(self._conanfile.generators_folder, MSBuildToolchain.filename)
+        if not os.path.exists(msbuildtoolchain_file):
```

Comment:
It is possible that some users want to inject only dependencies information, but they don't want to use the toolchain at all? Sounds very possible.

Reply:
Il would make little sense. It would be like allowing `CMake` build helper to not consume `conan_toolchain.cmake`.